File: paper/大论文较近版本/代码/MLIEM.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <windows.h>
int main()
{
	
	double run_time=0;

	int j=1;
	while(j<=2)
	{
		printf("Case:%d\n",j++);
		srand(time(NULL));
		int N,M;
		float u1,u2;
		float x[6];
		int i;
		int begintime=0,endtime=0;
		float p[5];
		float risk;
		float G1,G2;
		int pro=1;
		int next=11;
		int count=0;
	    float drr=0;
	    float fore=0;
	    float trend=0;
	    int send=rand()%2;
	    int res=rand()%2;
		
		printf("Please input N and M:");
		scanf("%d %d",&N,&M);
		printf("Please input u1 and u2:");
		scanf("%f %f",&u1,&u2);
		printf("Please input x[5]:\n");
		x[0]=0;
		for(i=1;i<=5;i++)
		{
			printf("x[%d]:",i);
			scanf("%f",&x[i]);
			fore = fore + x[i];
			if(x[i]>x[i-1])
			{
				count++;
			}
		}
		begintime=clock();	//计时开始
		/*
		_LARGE_INTEGER time_start;	//开始时间
		_LARGE_INTEGER time_over;	//结束时间
		double dqFreq;		//计时器频率
		LARGE_INTEGER f;	//计时器频率
		QueryPerformanceFrequency(&f);
		dqFreq=(double)f.QuadPart;
		QueryPerformanceCounter(&time_start);	//计时开始
		*/
		count = count-1;
		
		fore = fore/5;
		printf("fore:%f\n",fore);
		
		switch(count){
			case 0:trend=0.1;break;
			case 1:trend=0.3;break;
			case 2:trend=0.5;break;
			case 3:trend=0.7;break;
			case 4:trend=0.9;break;
		}
		printf("trend:%f\n",trend);
		fore = fore*trend;
		printf("new fore:%f\n",fore);
	    //p=rand()/(double)(RAND_MAX);
	    //a=rand()/(double)(RAND_MAX);
	    //s=rand()/(double)(RAND_MAX);
	    //e=rand()/(double)(RAND_MAX);
	    //m=rand()/(double)(RAND_MAX);
	    int tt;
	    i=0;
	    while(i<5){
	    	tt=rand()%5;
	    	switch(tt){
	    		case 0:break;
	    		case 1:p[i]=0.1;break;
	    		case 2:p[i]=0.35;break;
	    		case 3:p[i]=0.6;break;
	    		case 4:p[i]=0.85;break;
	    	}
	    	if(tt==0)
	    	{
	    		continue;
	    	}
	    	i++;
	    }
	    printf("p:%f a:%f s:%f e:%f m:%f\n",p[0],p[1],p[2],p[3],p[4]);
	    
	    risk = (1-p[0])*(1-p[1])*(1-p[2])*(1-p[3])*(1-p[4]);
	    printf("risk:%f\n",risk);
	    /* 
	    if(send==1 && res==1)
	    {
	    	printf("1 1\n");
	    	G1 = risk*(-N+M);
	    	G2 = risk*(3*N-M);
	    	printf("G1:%f G2:%f\n",G1,G2);
	    }
	    else if(send==1 && res==0)
	    {
	    	printf("1 0\n");
	    	G1 = risk*(-N);
	    	G2 = risk*(3*N);
	    	printf("G1:%f G2:%f\n",G1,G2);
	    }
	    else if(send==0 && res==1)
	    {
	    	printf("0 1\n");
	    	G1 = 0;
	    	G2 = 0;
	    	printf("G1:%f G2:%f\n",G1,G2);
	    }
	    else if(send==0 && res==0)
	    {
	    	printf("0 0\n");
	    	G1 = 0;
	    	G2 = 0;
	    	printf("G1:%f G2:%f\n",G1,G2);
	    }
	    else
	    {
	    	printf("Error!\n");
	    }*/
	    
	    
	    printf("1 1\n");
	    G1 = risk*(-N+M);
	    G2 = risk*(3*N-M);
	    printf("G1:%f G2:%f\n",G1,G2);
	    //drr = u1*fore + u2*G1;
	    //printf("drr:%f\n\n",drr);
	    
	    printf("1 0\n");
    	G1 = risk*(-N);
    	G2 = risk*(3*N);
    	printf("G1:%f G2:%f\n",G1,G2);
    	//drr = u1*fore + u2*G1;
	    //printf("drr:%f\n\n",drr);
	    
	    /*
	    QueryPerformanceCounter(&time_over);	//计时结束
		run_time = 1000000*(time_over.QuadPart-time_start.QuadPart)/dqFreq;
	    printf("\nrun_time：%fus\n",run_time);
	    */
	    endtime = clock();	//计时结束
		printf("\n\nRunning Time：%dms\n", endtime-begintime);
		printf("\n\n");
	}
	
	
	//乘以1000000把单位由秒化为微秒，精度为1000 000/（cpu主频）微秒
	
	return 0;

    
} 

File: paper/大论文较近版本/代码/MLIEM.py
from __future__ import print_function
import random

j=1;
while j<=2:
	print("Case:",j)
	j = j+1

	pro=1
	next=11
	count=0
	drr=0
	fore=0
	trend=0
	p = []
	x = []
	send = random.randint(0,1)
	res = random.randint(0,1)

	# print(send)
	# print(res)

	print("Please input N and M:")
	N = input()
	M = input()
	print("Please input u1 and u2:")
	u1 = input()
	u2 = input()
	print("Please input x[5]:")
	x.append(0)
	for i in range(1,6):
		print("x[",i,"]")
		t = input()
		x.append(t)
		fore = fore + x[i]
		if x[i]>x[i-1]:
			count = count+1;

	count = count -1
	print("count:",count)
	print("fore:",fore)

	fore = fore/5.0;
	print("fore:",fore)

	if count==0:
		trend = 0.1
	elif count==1:
		trend = 0.3
	elif count==2:
		trend = 0.5
	elif count==3:
		trend = 0.7
	elif count==4:
		trend = 0.9

	print("trend:",trend)

	fore = fore * trend;
	print("new fore:",fore)

	i=0
	while i<5:
		tt = random.randint(1,4)
		if tt==1:
			p.append(0.1)
		elif tt==2:
			p.append(0.35)
		elif tt==3:
			p.append(0.6)
		elif tt==4:
			p.append(0.85)
		i = i+1;

	print("p:",p[0],"  a:",p[1],"  s:",p[2],"  e:",p[3],"  m:",p[4])

	risk = (1-p[0]) * (1-p[1]) * (1-p[2]) * (1-p[3]) * (1-p[4])

	print("risk:",risk)

	# if send==1 and res==1:
	# 	print("1 1")
	# 	G1 = risk*(-N+M)
	# 	G2 = risk*(3*N)
	# 	print("G1:",G1,"  G2:",G2)
	# elif send==1 and res==0:
	# 	print("1 0")
	# 	G1 = risk*(-N)
	# 	G2 = risk*(3*N)
	# 	print("G1:",G1,"  G2:",G2)
	# elif send==0 and res==1:
	# 	print("0 1")
	# 	G1 = 0
	# 	G2 = 0
	# 	print("G1:",G1,"  G2:",G2)
	# elif send==0 and res==0:
	# 	print("0 0")
	# 	G1 = 0
	# 	G2 = 0
	# 	print("G1:",G1,"  G2:",G2)
	# else:
	# 	print("Error!")

	# drr = u1*fore + u2*G1
	# print("drr:",drr)



	
	print("1 1")
	G1 = risk*(-N+M)
	G2 = risk*(3*N)
	print("G1:",G1,"  G2:",G2)
	drr = u1*fore + u2*G1
	print("drr:",drr)


	print("1 0")
	G1 = risk*(-N)
	G2 = risk*(3*N)
	print("G1:",G1,"  G2:",G2)
	drr = u1*fore + u2*G1
	print("drr:",drr)



	print("0 1")
	G1 = 0
	G2 = 0
	print("G1:",G1,"  G2:",G2)
	drr = u1*fore + u2*G1
	print("drr:",drr)



	print("0 0")
	G1 = 0
	G2 = 0
	print("G1:",G1,"  G2:",G2)
	drr = u1*fore + u2*G1
	print("drr:",drr)

	


	

